.github/workflows/sonarcloud.yml

```diff
@@ -0,0 +1,40 @@
+name: sonarcloud
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+    - main
+
+jobs:
+
+  sonarcloud:
+    name: SonarCloud
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.10'
+      - name: Python info
+        shell: bash -l {0}
+        run: |
+          which python3
+          python3 --version
```

Comment:
There is the package `xESMF` in the [environment.yaml](https://github.com/EcoExtreML/zampy/blob/54439a0e923da7a1ddeac813e83bc8e7088fa1dc/environment.yml#L7) that cannot be installed via pip and thus not in the [pyproject.toml](https://github.com/EcoExtreML/zampy/blob/54439a0e923da7a1ddeac813e83bc8e7088fa1dc/pyproject.toml#L50). So, we need to first create the environment and then install the package e.g. 
```
name: Set up Python
        uses: conda-incubator/setup-miniconda@v2
        with:
          activate-environment: zampy
          environment-file: environment.yml
          mamba-version: "*"
          python-version: "3.10"
          miniconda-version: "latest"
          channels: conda-forge
```